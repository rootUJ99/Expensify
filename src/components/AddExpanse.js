import React from 'react';
import ReactDOM from 'react-dom';
const AddExpanse = () => (
    <div>
        this is add page
    </div>
);
export default AddExpanse;