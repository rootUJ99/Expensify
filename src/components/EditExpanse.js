import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
export class EditExpanse extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
        console.log('updated', expense);
    }
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <div className="page-header" >
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    className="button2 button2--remove"
                    onClick={this.onRemove}
                > Remove Expense</button>
            </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};
const mapDispathToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispathToProps)(EditExpanse);