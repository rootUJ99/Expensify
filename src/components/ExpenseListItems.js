import React from 'react';
import {Link} from 'react-router-dom';
const ExpenseListItem = ({ id, dispatch, description, amount, createdAt})=>(
    <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <h4>{amount}-{createdAt}</h4>
       
    </div>
);

export default ExpenseListItem;
