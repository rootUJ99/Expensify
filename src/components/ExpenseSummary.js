import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import {Link} from 'react-router-dom';
export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div className="page-header">
        <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount} {expenseWord}</span> totalling <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions"></div>
                <Link className="button2" to="/create">Add Expense</Link>
        </div>
        </div>
    );
}
const mapStateToProps=(state)=>{
    const visiblityExpenses = selectExpenses(state.expenses,state.filters);
    return{
        expenseCount:visiblityExpenses.length,
        expensesTotal:selectExpensesTotal(visiblityExpenses)
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
