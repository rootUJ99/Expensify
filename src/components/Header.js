import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="active-nav" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="active-nav">Create</NavLink>
        <NavLink to="/edit" activeClassName="active-nav">Edit</NavLink>
    </header>
);
export default Header;