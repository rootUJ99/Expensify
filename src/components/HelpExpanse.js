import React from 'react';
import ReactDOM from 'react-dom';
const HelpExpanse = () => (
    <div>
        this is help
    </div>
);
export default HelpExpanse;