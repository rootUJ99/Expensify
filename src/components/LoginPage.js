import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
        <h1>Expensify Application</h1>
            <button className="button" onClick={startLogin}>Login</button>
        </div>
    </div>
)
const matchDispatchProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});
export default connect(undefined, matchDispatchProps)(LoginPage);