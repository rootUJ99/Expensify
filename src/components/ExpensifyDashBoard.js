import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListItems from './ExpenseListItems';
import ExpenseListFilters from './ExpenseListFilters';
const ExpensifyDashBoard = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
export default ExpensifyDashBoard;