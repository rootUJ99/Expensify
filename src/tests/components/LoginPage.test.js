import React from 'react';
import { shallow } from 'enzyme';
import {LoginPage} from '../../components/LoginPage';
test('testing login form',()=>{
    const wrapper= shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
})
test('should log in on button click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});