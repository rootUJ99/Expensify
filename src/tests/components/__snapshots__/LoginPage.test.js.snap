// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing login form 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1>
      Expensify Application
    </h1>
    <button
      className="button"
    >
      Login
    </button>
  </div>
</div>
`;
